autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -231,46 +232,52 @@ const Schedule: React.FC<RouteComponentProps> = () => {
     </div>
   ));
 
-  // build each day based on schedule
-  function getMeetingsForDay(day: number): Meeting[] {
-    // meetingDays = MTWRFSU
-    // day = MTWRF
-    return schedule.filter((meeting) => meeting.meetingDays[day]);
-  }
-  function getAvailabilityForDay(day: number): Availability[] {
-    return availabilityList.filter((avl) => avl.dayOfWeek === day);
-  }
-  function renderMeeting(meeting: Meeting): JSX.Element {
-    const colors = ['#500000', '#733333', '#966666', '#b99999', '#dccccc'];
-    return (
-      <MeetingCard
-        meeting={meeting}
-        bgColor={colors[uniqueSections.indexOf(meeting.section.id)]}
-        key={meeting.id}
-        firstHour={FIRST_HOUR}
-        lastHour={LAST_HOUR}
-      />
+  // let's see if useMemo reduces our render time
+  const meetingsForDays = React.useMemo(() => {
+    const uniqueSections = [...new Set(schedule.map((mtg: Meeting) => mtg.section.id))];
+    // build each day based on schedule
+    function getMeetingsForDay(day: number): Meeting[] {
+      // meetingDays = UMTWRFS
+      // day = MTWRF
+      return schedule.filter((meeting) => meeting.meetingDays[day]);
+    }
+    function renderMeeting(meeting: Meeting): JSX.Element {
+      const colors = ['#500000', '#733333', '#966666', '#b99999', '#dccccc',
+        '#871b1e', '#9f494b', '#b77678', '#9a1d26', 'c2777d'];
```

Comment:
could you add this amount of sections/meetings to the test data so we can see the different colors?

Reply:
Done. Also added a test for it in `ui/Schedule.test.tsx`